integration-tests/tests/integrations.rs

```diff
@@ -3577,14 +3422,153 @@ mod slow {
+                    &get_base_address!(client, "transparent"),
+                    28000,
+                    None,
+                )])
+                .await
+                .unwrap();
+        }
+    }
+
+    #[tokio::test]
+    async fn load_wallet_from_v26_dat_file() {
+        // We test that the LightWallet can be read from v26 .dat file
+        // Changes in version 27:
+        //   - The wallet does not have to have a mnemonic.
+        //     Absence of mnemonic is represented by an empty byte vector in v27.
+        //     v26 serialized wallet is always loaded with `Some(mnemonic)`.
+        //   - The wallet capabilities can be restricted from spending to view-only or none.
+        //     We introduce `Capability` type represent different capability types in v27.
+        //     v26 serialized wallet is always loaded with `Capability::Spend(sk)`.
+
+        // A testnet wallet initiated with
+        // --seed "chimney better bulb horror rebuild whisper improve intact letter giraffe brave rib appear bulk aim burst snap salt hill sad merge tennis phrase raise"
+        // with 3 addresses containing all receivers.
+        // including orchard and sapling transactions
+        let data = include_bytes!("zingo-wallet-v26.dat");
+
+        load_wallet_from_data_and_assert(data, 0, 3).await;
+    }
+
+    #[tokio::test]
+    async fn load_wallet_from_v26_2_dat_file() {
+        // We test that the LightWallet can be read from v26 .dat file
+        // Changes in version 27:
+        //   - The wallet does not have to have a mnemonic.
+        //     Absence of mnemonic is represented by an empty byte vector in v27.
+        //     v26 serialized wallet is always loaded with `Some(mnemonic)`.
+        //   - The wallet capabilities can be restricted from spending to view-only or none.
+        //     We introduce `Capability` type represent different capability types in v27.
+        //     v26 serialized wallet is always loaded with `Capability::Spend(sk)`.
+
+        // A testnet wallet initiated with
+        // --seed "chimney better bulb horror rebuild whisper improve intact letter giraffe brave rib appear bulk aim burst snap salt hill sad merge tennis phrase raise"
+        // with 3 addresses containing all receivers.
+        // including orchard and sapling transactions
+        let data = include_bytes!("zingo-wallet-v26-2.dat");
+
+        load_wallet_from_data_and_assert(data, 10177826, 1).await;
+    }
+
+    #[tokio::test]
+    async fn load_wallet_from_v28_dat_file() {
+        // We test that the LightWallet can be read from v28 .dat file
+        // --seed "chimney better bulb horror rebuild whisper improve intact letter giraffe brave rib appear bulk aim burst snap salt hill sad merge tennis phrase raise"
+        // with 3 addresses containing all receivers.
+        let data = include_bytes!("zingo-wallet-v28.dat");
+
+        load_wallet_from_data_and_assert(data, 10342837, 3).await;
+    }
```

Comment:
i think these were already merged right? can you merge/rebase and push to clear up the git diff?